src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1661,8 +1661,8 @@ private void SetRequestHandleClientCertificateOptions(SafeWinHttpHandle requestH
             }
 
             // Reduce the set of certificates to match the proper 'Client Authentication' criteria.
-            certs = certs.Find(X509FindType.FindByKeyUsage, X509KeyUsageFlags.DigitalSignature, true);
-            certs = certs.Find(X509FindType.FindByApplicationPolicy, ClientAuthenticationOID, true);
+            certs = certs.Find(X509FindType.FindByKeyUsage, X509KeyUsageFlags.DigitalSignature, false);
+            certs = certs.Find(X509FindType.FindByApplicationPolicy, ClientAuthenticationOID, false);
```

Comment:
Note that Find returns a cloned (DuplicateHandle()d) instance, so two calls to find will result in every result from the first one to be sent off to the GC (and the SafeHandles to be sent off to the Finalizer).

So this would be more efficiently written as a foreach; and if cloned output was desired in the end, calling find on a collection containing one item would be a way of accomplishing that.

But that's just some friendly perf advice, certainly not a must-fix level of bad.